src/ServerScriptService/InstanceComponents/Structures/Observer.lua

```diff
@@ -0,0 +1,35 @@
+---@diagnostic disable-next-line: undefined-global
+local Players = game:GetService("Players")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local RunService = game:GetService("RunService")
+
+local Settings = require(ReplicatedStorage.Common.Config.Terrain)
+local MinimumRange = Settings.MinimumRange
+
+local Observer = {}
+Observer.__index = Observer
+
+function Observer.new(instance)
+	local self = setmetatable({}, Observer)
+	self.Instance = instance
+
+    RunService.Heartbeat:Connect(function()
```

Comment:
There's a memory leak here! This connection will persist after the observer is destroyed!

Reply:
You haven't resolved the memory leak. You're still leaking the table and the connection. The function is still being called every heatbeat.